File: Tang/Tang/Content/Module/Login/ViewCtrl/TangLoginNaviCtrl.m
//
//  TangLoginNaviCtrl.m
//  Tang
//
//  Created by wwwbbat on 2017/9/16.
//  Copyright © 2017年 tiny. All rights reserved.
//

#import "TangLoginNaviCtrl.h"
#import "TangLoginViewCtrl.h"

@interface TangLoginNaviCtrl ()

@end

@implementation TangLoginNaviCtrl

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    ygweakify(self);
    TangLoginViewCtrl *vc = [self loginViewCtrl];
    vc.didLogined = ^{
        ygstrongify(self);
        if (self.didLogined) {
            self.didLogined();
        }
    };
}

- (TangLoginViewCtrl *)loginViewCtrl
{
    UIViewController *vc = self.viewControllers.firstObject;
    if ([vc isKindOfClass:[TangLoginViewCtrl class]]) {
        return (TangLoginViewCtrl *)vc;
    }
    NSAssert(NO, @"导航控制器层级错误！");
    return nil;
}

@end

File: Tang/Tang/AppCode/Module/Login/Ctrls/TangLoginNaviCtrl.swift
//
//  TangLoginNaviCtrl.swift
//  Tang
//
//  Created by Tiny on 2018/2/8.
//  Copyright © 2018年 tiny. All rights reserved.
//

import UIKit

class TangLoginNaviCtrl: YGBaseNaviCtrl {
    
    var didLogined : (()->Void)?

    override func viewDidLoad() {
        super.viewDidLoad()

        loginViewCtrl?.didLogined = {
            [unowned self] in
            if let block = self.didLogined {
                block()
            }
        }
    }

    private var loginViewCtrl : TangLoginViewCtrl? {
        get {
            if let vc = viewControllers.first as? TangLoginViewCtrl {
                return vc
            }
            assert(false, "视图层级错误")
            return nil
        }
    }
    

}
